de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/util/VirSatWaitForProjectBuilder.java

```diff
@@ -0,0 +1,185 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.swtbot.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.IJobChangeListener;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
+
+import de.dlr.sc.virsat.swtbot.test.ASwtBotTestCase;
+
+/**
+ * This Condition waits for the Workspace Builder to execute
+ *
+ */
+public class VirSatWaitForProjectBuilder extends DefaultCondition implements IJobChangeListener {

```

Comment:
just wondering, can we maybe use (or extend) `org.eclipse.swtbot.eclipse.finder.waits.WaitForJobs`?

Reply:
We used to use the WaitForJobs. This is an improvement which counts the jobs being called and the jobs being stopped.